### DISCUSSION TITLE
Improve C# 'switch' statement

### DISCUSSION THREAD

**Discussion by The-Futurist** [dotnet/csharplang#283] [General]
Consider improving the **switch** statement in C#.

Specifically remove the restriction that each **case** clause must specify only constants and allow arbitrary expressions.

**Comment by svick** [comment]
Do you have an example where this would be useful?

**Comment by alrz** [comment]
VB supports this I don't think that is a good thing. All side effects would run up to the matching `case`.

**Comment by DavidArno** [comment]
This suggestion is already covered by the numerous proposals to expand pattern matching.
